File: app/src/main/java/quintype/com/templatecollectionwithrx/models/NavMenuGroup.java
package quintype.com.templatecollectionwithrx.models;

import android.content.Context;
import android.os.Parcel;
import android.os.Parcelable;

import com.bignerdranch.expandablerecyclerview.Model.ParentListItem;

import java.util.ArrayList;
import java.util.List;

import quintype.com.templatecollectionwithrx.R;

public class NavMenuGroup implements ParentListItem, Parcelable {

    public static final Creator<NavMenuGroup> CREATOR = new Creator<NavMenuGroup>() {
        @Override
        public NavMenuGroup createFromParcel(Parcel source) {
            return new NavMenuGroup(source);
        }

        @Override
        public NavMenuGroup[] newArray(int size) {
            return new NavMenuGroup[size];
        }
    };
    //to set the right tab in the viewPager
    int position;
    boolean selected;
    //the parent menu item
    private NavMenu menuItem;
    //the child menu items
    private List<NavMenu> mSubsections = new ArrayList<>();
    //to differentiate dummy elements
    private String dummyValue = null;

    public NavMenuGroup() {
    }

    protected NavMenuGroup(Parcel in) {
        this.position = in.readInt();
        this.menuItem = in.readParcelable(NavMenu.class.getClassLoader());
        this.selected = in.readByte() != 0;
        this.mSubsections = in.createTypedArrayList(NavMenu.CREATOR);
        this.dummyValue = in.readString();
    }

    public String getDummyValue() {
        return dummyValue;
    }

    public void setDummyValue(String dummyValue) {
        this.dummyValue = dummyValue;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }

    public void addSubsection(NavMenu subsection) {
        mSubsections.add(subsection);
    }

    public NavMenu getMenuItem() {
        return menuItem;
    }

    public void setMenuItem(NavMenu menuItem) {
        this.menuItem = menuItem;
    }

    public String getName(Context context) {
        if (menuItem.id().equals(NavMenu.HOME.id())) {
            return context.getResources().getString(R.string.home_title);
        }else{
            return menuItem.title();
        }
    }

    @Override
    public List<NavMenu> getChildItemList() {
        return mSubsections;
    }

    @Override
    public boolean isInitiallyExpanded() {
        return false;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.position);
        dest.writeParcelable(this.menuItem, flags);
        dest.writeByte(this.selected ? (byte) 1 : (byte) 0);
        dest.writeTypedList(this.mSubsections);
        dest.writeString(this.dummyValue);
    }

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }
}

File: app/src/main/java/quintype/com/templatecollectionwithrx/models/NavMenuGroup.kt
package quintype.com.templatecollectionwithrx.models

import android.content.Context
import android.os.Parcel
import android.os.Parcelable
import com.bignerdranch.expandablerecyclerview.Model.ParentListItem
import quintype.com.templatecollectionwithrx.R

class NavMenuGroup() : ParentListItem, Parcelable {
    var position: Int = 0
    var selected: Boolean = false
    public var menuItem: NavMenu? = null
    private var mSubsections = arrayListOf(NavMenu())
    public var dummyValue: String? = null

    constructor(parcel: Parcel) : this() {
        position = parcel.readInt()
        menuItem = parcel.readParcelable(NavMenu::class.java.classLoader)
        selected = !parcel.readByte().equals(0)
        mSubsections = parcel.createTypedArrayList(NavMenu.CREATOR)
        dummyValue = parcel.readString()
    }

    fun getName(context: Context): String? {
        return if (menuItem?.id().equals(NavMenu.HOME.id())) {
            context.getResources().getString(R.string.home_title)
        } else {
            menuItem?.title()
        }
    }

    override fun getChildItemList(): MutableList<NavMenu> {
        return mSubsections
    }

    override fun isInitiallyExpanded(): Boolean {
        return false
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(this.position)
        parcel.writeParcelable(this.menuItem, flags)
        parcel.writeByte(if (this.selected) 1.toByte() else 0.toByte())
        parcel.writeTypedList(this.mSubsections)
        parcel.writeString(this.dummyValue)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<NavMenuGroup> {
        override fun createFromParcel(parcel: Parcel): NavMenuGroup {
            return NavMenuGroup(parcel)
        }

        override fun newArray(size: Int): Array<NavMenuGroup?> {
            return arrayOfNulls(size)
        }
    }
}